datafusion/core/src/execution/context/parquet.rs

```diff
@@ -94,6 +96,18 @@ impl SessionContext {
     ) -> Result<()> {
         plan_to_parquet(self.task_ctx(), plan, path, writer_properties).await
     }
+
+    /// Registers a Parquet [`EncryptionFactory`] with an associated unique identifier.
+    /// If an encryption factory with the same identifier was already registered, it is replaced and returned.
+    #[cfg(feature = "parquet_encryption")]
```

Comment:
I suggest we  keep the API surface area lower by only adding the API to the runtime_env rather than also adding this wrapper to SessionContext

So users would have to do
```rust
ctx.runtime_env()
  .register_parquet_encryption_factory(id, encryption_factory)
```

Rather than what you have here
```rust
ctx.register_parquet_encryption_factory(id, encryption_factory)
```


I don't feel strongly about this

Reply:
:+1: that sounds sensible to me